ingestion/src/metadata/ingestion/source/database/bigquery/connection.py

```diff
@@ -55,6 +55,10 @@ def get_connection_url(connection: BigQueryConnection) -> str:
     environment variable when needed
     """
 
+    # If projectId is defined in config, use it by default
+    if connection.projectId:
+        return f"{connection.scheme.value}://{connection.projectId}"
```

Comment:
We should add a validation where, if it's isinstance of GcpCredentialsPath and if projectId is provided, we use that

Reply:
Actually, taking a look at it, let's modify the json schema, we can move projectId within the GcpCredentialsPath json schema, given that the scope of the above is only for the when the user provides a path to service account file